src/evaluators/BinaryExpression.js

```diff
@@ -148,12 +148,14 @@ export function computeBinary(
   loc?: ?BabelNodeSourceLocation
 ): Value {
   // partial evaluation shortcut for a particular pattern
-  if (op === "==" || op === "===" || op === "!=" || op === "!==") {
+  if (realm.useAbstractInterpretation && (op === "==" || op === "===" || op === "!=" || op === "!==")) {
     if (
       (!lval.mightNotBeObject() && (rval instanceof NullValue || rval instanceof UndefinedValue)) ||
       ((lval instanceof NullValue || lval instanceof UndefinedValue) && !rval.mightNotBeObject())
-    )
+    ) {
+      if (lval.isIntrinsic() || rval.isIntrinsic()) return AbstractValue.createFromBinaryOp(realm, op, lval, rval, loc);
```

Comment:
Not sure what the intention here is. Are you after object values, or just abstract values that might be aliased? In general, this new behavior seems too crude.
- is the point h

Reply:
As discussed earlier, we need a representation for intrinsic objects that can be null or undefined. The realization exploited by the request is that if we assume that any intrinsic object may be null or undefined, then the only thing in the code that needs to change is this bit.

Reply:
I see. Still, allowing for an `ObjectValue` to be potentially null or undefined is quite surprising. Do you consider this a quick hack to unblock the React Compiler, or does this permanently alter the meaning of `ObjectValue`? (Will be important for any future simplification rules or even a formal encoding for an SMT solver.)

Can you leave behind a longer comment in the code about this?

Reply:
I'll add a comment. Note that only intrinsic object values are affected by this.